### PR TITLE
Dbless

### PR DESCRIPTION
**Pull request by kamiiiel**
## Issue

https://gravitee.atlassian.net/browse/APIM-303

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-fkdueroxpe.chromatic.com)

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -963,6 +966,7 @@ gateway:
   #     request:
   #       header: X-Gravitee-Request-Id
   reporters:
+    type: "elasticsearch" # if set to "none" no report will be submitted
```
Comment (sebdevaux) [inline helm/values.yaml]: Why did we change this parameter?
`    reporters:
      type: elasticsearch
      # Elasticsearch reporter
      elasticsearch:
        enabled: true
        endpoints:
          - http://graviteeio-apim-elasticsearch.default.svc.cluster.local:9200
        index_mode: daily
        pipeline:
          plugins:
            ingest: geoip, user_agent
        index: gravitee
        bulk:
          actions: 1000           # Number of requests action before flush
          flush_interval: 5       # Flush interval in seconds
      type:
        elasticsearch`

Now we have a duplicate key in the configmap.yaml

**Comment by sebdevaux** [comment]
Could you please add Tests with dbLess: true?

**Review by sebdevaux — APPROVED** [review]
Ok for the Helm Chart part

File: gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-jwt/src/main/java/io/gravitee/gateway/security/jwt/policy/CheckSubscriptionPolicy.java [hunk 2]
```diff
@@ -48,10 +49,18 @@ public void execute(PolicyChain policyChain, ExecutionContext executionContext)
         final String api = (String) executionContext.getAttribute(ExecutionContext.ATTR_API);
         final String clientId = (String) executionContext.getAttribute(CONTEXT_ATTRIBUTE_CLIENT_ID);
         final String plan = (String) executionContext.getAttribute(ExecutionContext.ATTR_PLAN);
+        final String repositoryType = (String)  executionContext.getAttribute(CONTEXT_REPOSITORY_TYPE);
 
         executionContext.request().metrics().setSecurityType(JWT);
         executionContext.request().metrics().setSecurityToken(clientId);
 
+        // DB Less mode
+        if ("none".equals(repositoryType)) {
```
Comment (guillaumelamirand) [inline gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-jwt/src/main/java/io/gravitee/gateway/security/jwt/policy/CheckSubscriptionPolicy.java]: Shouldn't we avoid having such hard coded behavior depending on the repository inside a Subscription process?

**Review by guillaumelamirand — CHANGES_REQUESTED** [review]
I am concerned about how noop repositories are used which will require implementing a new empty repository for any new repository added. I am also wondering how the distributed sync (using redis) will be impacted by such changes.

Could you also provide documentation about the analysis explaining the context, technical approach, and expectation from the customer's point of view?

File: gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-jwt/src/main/java/io/gravitee/gateway/security/jwt/policy/CheckSubscriptionPolicy.java (hunk 2 above)
Comment (kamiiiel) [reply gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-jwt/src/main/java/io/gravitee/gateway/security/jwt/policy/CheckSubscriptionPolicy.java]: I made some change a moved the logic to the subscription service

Comment (kamiiiel) [reply gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-jwt/src/main/java/io/gravitee/gateway/security/jwt/policy/CheckSubscriptionPolicy.java]: @guillaumelamirand Are you OK with this new changes?

**Review by a-cordier — APPROVED** [review]

**Pull request merged**